include/exadg/acoustic_conservation_equations/spatial_discretization/spatial_operator.cpp

```diff
@@ -247,14 +247,28 @@ void
 SpatialOperator<dim, Number>::serialize_vectors(
   std::vector<BlockVectorType const *> const & block_vectors) const
 {
+  // Write deserialization parameters. These do not change during the simulation, but the data are
+  // small and we want to make sure to overwrite them.
+  DeserializationParameters deserialization_parameters;
+  deserialization_parameters.degree_u               = param.degree_u;
+  deserialization_parameters.degree_p               = param.degree_p;
+  deserialization_parameters.mapping_degree         = param.mapping_degree;
+  deserialization_parameters.consider_mapping_write = param.restart_data.consider_mapping_write;
+  deserialization_parameters.triangulation_type     = param.grid.triangulation_type;
+  write_deserialization_parameters(mpi_comm,
+                                   param.restart_data.directory,
+                                   param.restart_data.filename,
+                                   deserialization_parameters);
+
+  // Attach vectors to triangulation and serialize.
   std::vector<dealii::DoFHandler<dim> const *> dof_handlers(2);
   dof_handlers.at(block_index_velocity) = &this->get_dof_handler_u();
   dof_handlers.at(block_index_pressure) = &this->get_dof_handler_p();
 
   std::vector<std::vector<VectorType const *>> vectors_per_dof_handler =
     get_vectors_per_block<VectorType const, BlockVectorType const>(block_vectors);
 
-  if(param.restart_data.consider_mapping)
+  if(param.restart_data.consider_mapping_write)
```

Comment:
switch if the mapping should be attached to the triangulation as a displacement vector or not.